src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -98,6 +128,22 @@ public class JsonInputFormatter : TextInputFormatter
             }
 
             var request = context.HttpContext.Request;
+
+            if (request.Body.CanSeek)
+            {
+                request.Body.Seek(0L, SeekOrigin.Begin);
```

Comment:
🆙 📅  Should we do this or the layer which buffered the request should rewind it back?